eldritch/events.py

```diff
@@ -568,22 +572,26 @@ def resolve(self, state):
     if self.drawn is None:
       self.drawn = []
       deck = getattr(state, self.deck)
-      i = 0
-      decksize = len(deck)
-      while len(self.drawn) < self.draw_count:
-        i += 1
-        if not deck :
-          break
-        top = deck.popleft()
-        if self.target_type is None or isinstance(top, self.target_type):
-          self.drawn.append(top)
-        else:
-          deck.append(top)
-
-        if i >= decksize:
-          break
-      # TODO: is there a scenario when the player can go insane/unconscious before they
-      # successfully pick a card?
+      if self.draw_count == 'all':
```

Comment:
Instead of "all", you can use float("inf"). Avoids needing to special-case "all".

Reply:
Fixed here, which should at least make merging the Draw changes simpler.